core/src/main/java/org/apache/calcite/plan/AbstractRelOptPlanner.java

```diff
@@ -96,6 +101,11 @@ protected AbstractRelOptPlanner(RelOptCostFactory costFactory,
     // these types, but some operands may use them.
     classes.add(RelNode.class);
     classes.add(RelSubset.class);
+
+    if (LOGGER.isDebugEnabled()) {
+      this.ruleAttemptsListener = new RuleAttemptsListener();
```

Comment:
Should it use its own logging category instead?

Then it can be a trace level for its own category.

Reply:
I don't think we need a separate category just for the rule attempts. IMO it makes more sense to put them in DEBUG level log, together with the best plan and provenance map. And the verbose rule firing and transformation messages should really be TRACE.